### ISSUE TITLE
Pulseaudio support?

### ISSUE THREAD

**Issue by ThYpHo0n** [mikebrady/shairport-sync#266]
Hey,

first of all thanks for this great project! I got it working very well to stream via Airplay to my BT speakers. But I would really like to get it working with Pulseaudio because I want to have multiple audio sources mix them together with priority (e.g. Amazon Echo API > AirPlay > MPD with nice fading effect for each switch) and process them through to multiple audio devices (a set BT speaker in different rooms). Yeah it's a quite ambitious goal :) I can't see a good possibility in Alsa to accomplish this and would prefer to use Pulseaudio but can't get it working so far with shairport-sync. The only way I didn't tested so far was the fifo-pipe output which I may can get into Pulseaudio somehow (have to research on that). The sideproblem with the lack of Pulseaudio support is that I have to use Bluez 4.x instead of 5 because Bluez 5 dropped Alsa support (related: https://github.com/mikebrady/shairport-sync/issues/200#issuecomment-192982716 )

TL;DR: Is there a working Pulseaudio integration planned? Or did I missed it and I can't just configure it properly for Pulseaudio?

**Comment by mikebrady** [comment]
Thanks for the post, and it's great that you've had success with Shairport Sync.

There are no plans to do PulseAudio integration at present. This is partly driven by a need to focus on getting the software to work really well on `alsa`. There is a PulseAudio backend in Shairport Sync right now, but it is based on the simple interface and is completely unsuitable, IMHO, for further development. A fully-featuredPulseAudio back end is needed, and it might even require changes to Shairport Sync's backend arrangements. The really crucial thing is for PulseAudio to be able to give a real-time and reliable indication of latency to Shairport Sync. I think it would be quite a lot of effort to get that functionality, but I'm not a PulseAudio guy.

**Comment by mikebrady** [comment]
Hi there. I'm closing this issue, if you don't mind. Please feel free to reopen it if there are any developments.

**Comment by biogerm** [comment]
Hi @ThYpHo0n @mikebrady !

It seems the latest raspbian not only dropped alsa support because upgraded to bluz5, but it also upgraded alsa that is not compatible with bluez4. I tried the method mentioned [here](https://github.com/mikebrady/shairport-sync/issues/200) but it reports something missing from ALSA.

Do you guys have any idea how to get alsa working again?

**Comment by mikebrady** [comment]
Hi there. Raspbian has not dropped ALSA support. I do not know about the status of Bluetooth and ALSA though – I've never had any success with it.

**Comment by ThYpHo0n** [comment]
Moin @biogerm 
you can pin your packages to specific version to get it working but I wouldn't recommend this.
I've decided to not use Bluez together with Alsa anymore because it's not suitable for multiroom audio (the BT stack just isn't designed for that). My queston for you would be why you need to go with the Alsa/Bluez combination?
I'm using a filesink/unix pipe and distribute that via https://github.com/badaix/snapcast to multiple Raspis (Raspian) which works flawless.

**Comment by mikebrady** [comment]
That Snapcast looks interesting alright – thanks for the information.

**Comment by mikebrady** [comment]
Just FYI, PulseAudio support is now in the development branch of Shairport Sync.

**Comment by VidJa** [comment]
Hey nice, I've been looking into this. I can stream my iphone to my linux homeserver and get sound on the attached speakers (its running ubuntu 16.04) However I would like to pass it on to my sonos bar if possible. (the sonos can be used as dlna endpoint apparently, see: https://en.community.sonos.com/troubleshooting-228999/basic-upnp-dlna-question-41672) 
it would be awesome to set this up, but since I'm new to this subject it would be nice to get some hints on how this could work

**Comment by JustinYuan94** [comment]
> Hey nice, I've been looking into this. I can stream my iphone to my linux homeserver and get sound on the attached speakers (its running ubuntu 16.04) However I would like to pass it on to my sonos bar if possible. (the sonos can be used as dlna endpoint apparently, see: https://en.community.sonos.com/troubleshooting-228999/basic-upnp-dlna-question-41672)
> it would be awesome to set this up, but since I'm new to this subject it would be nice to get some hints on how this could work

Hi, I think you can use airconnect.
